back/engines/commercial/report_builder/app/controllers/report_builder/web_api/v1/reports_controller.rb

```diff
@@ -44,6 +44,42 @@ def create
           render json: serialize_report(report), status: :created
         end
 
+        def copy
+          # We wrap the whole report creation in a transaction to ensure that the layout
+          # duplication (that is persisted directly — see {ContentBuilder::Layout#duplicate!}
+          # is rolled back if the report copy fails.
+          report_copy = ActiveRecord::Base.transaction do
+            copy = report.dup.tap do |copy_|
+              copy_.name = generate_copy_name(report.name)
+              copy_.owner_id = current_user.id
+              # Copies are never associated with a phase, even if the source report was.
+              # That's because, currently, there can be only one report per phase.
+              copy_.phase_id = nil
+              copy_.visible = false
+              copy_.layout = report.layout.duplicate!
+            end
+
+            authorize(copy)
+            side_fx_service.before_create(copy, current_user)
+
+            copy.save!
+            # Update self-references in the layout to point to the new report. The report
+            # copy must be saved before this operation, because it needs an ID.
+            copy.layout.update(craftjs_json: JSON.parse(
```

Comment:
```suggestion
            copy.layout.update!(craftjs_json: JSON.parse(
```
It seems there's no validation on `craftjs_json`, so it should never fail, but just in case new validations are introduced, it could be a good way to learn about errors early.